### DISCUSSION TITLE
Import a folder of gpx files into a My Places/Tracks folder

### DISCUSSION THREAD

**Discussion by tmusolf** [osmandapp/OsmAnd#19837] [Ideas]
In addition to importing individual gpx files into OSMAnd I would like to be able to import a folder of gpx files.  

Selecting a folder would import all of the gpx files found in the folder into a My Places/Tracks folder of the same name. 

If a folder of that name does not exist in My Places/Tracks it would be created.

If the folder already exists in My Places/tracks you could be given two options:

1. Delete the contents of the existing My Places/Tracks folder before importing the new gpx files.  This would make the source and imported folder contents identical.
2. Import the gpx files into the folder, overwriting any existing gpx files of the same name.  This would merge the source and imported folder contents.

Importing an entire folder of gpx files is a big time saver when you have many gpx files to update or bring into OSMAnd.  For example, I have many folders of gpx files, one for each trip I am planning.  Each folder may contain a 100 different gpx files of tracks and waypoints for that particular trip.  These gpx files are created on PC based tools and change over time.  At several points I may want to update the entire folder of gpx files that have been previously imported into OSMAnd.  Doing this one at a time is time consuming and error prone.

**Comment by jac486** [comment]
Hi. What I do is change the default storage folder in Settings -> OsmAnd Settings -> Data storage folder -> Set manually.

I let all the maps download there, and inside the Tracks folder I copy the tracks I want to import, keeping the folder structure I already have.

**Comment by DmitryAlexei** [comment]
Have you tried to zip your folder and then change .zip > .osf extention and open that file with OsmAnd?
